app/models/goal.py

```diff
@@ -2,4 +2,21 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement = True)
+    title = db.Column(db.String)
+    tasks = db.relationship("Task", back_populates="goal", lazy = True)
+
+    def to_dict(self):   
+        #tasks = [task.to_dict() for task in self.tasks]
+        return {
+            "id" : self.goal_id,
+            "title" : self.title
+            
+        }
+    def to_dict_with_tasks(self):
+        tasks = [task.to_dict_with_goal() for task in self.tasks]
+        return {
+            "id" : self.goal_id,
+            "title" : self.title,
+            "tasks" : tasks
+        }
```

Comment:
Yay, helper methods! However, these two are very, very similar. For the one with tasks, you can call the regular `to_dict` and then just add the lists of tasks to the dictionary that's returned.